frame/multisig/src/lib.rs

```diff
@@ -0,0 +1,550 @@
+use codec::{Encode, Decode};
+use sp_io::hashing::blake2_256;
+use frame_support::{decl_module, decl_event, decl_error, decl_storage, Parameter, ensure, RuntimeDebug};
+use frame_support::{traits::{Get, ReservableCurrency, Currency, Filter, FilterStack, ClearFilterGuard},
+	weights::{Weight, GetDispatchInfo, DispatchClass, FunctionOf, Pays},
+	dispatch::{DispatchResultWithPostInfo, DispatchErrorWithPostInfo, PostDispatchInfo},
+};
+use frame_system::{self as system, ensure_signed};
+use sp_runtime::{DispatchError, DispatchResult, traits::Dispatchable};
+
+mod tests;
+mod benchmarking;
+
+type BalanceOf<T> = <<T as Trait>::Currency as Currency<<T as frame_system::Trait>::AccountId>>::Balance;
+
+/// Configuration trait.
+pub trait Trait: frame_system::Trait {
+	/// The overarching event type.
+	type Event: From<Event<Self>> + Into<<Self as frame_system::Trait>::Event>;
+
+	/// The overarching call type.
+	type Call: Parameter + Dispatchable<Origin=Self::Origin, PostInfo=PostDispatchInfo>
+		+ GetDispatchInfo + From<frame_system::Call<Self>>;
+
+	/// The currency mechanism.
+	type Currency: ReservableCurrency<Self::AccountId>;
+
+	/// The base amount of currency needed to reserve for creating a multisig execution.
+	///
+	/// This is held for an additional storage item whose value size is
+	/// `4 + sizeof((BlockNumber, Balance, AccountId))` bytes.
+	type DepositBase: Get<BalanceOf<Self>>;
+
+	/// The amount of currency needed per unit threshold when creating a multisig execution.
+	///
+	/// This is held for adding 32 bytes more into a pre-existing storage value.
+	type DepositFactor: Get<BalanceOf<Self>>;
+
+	/// The maximum amount of signatories allowed in the multisig.
+	type MaxSignatories: Get<u16>;
+
+	/// Is a given call compatible with the proxying subsystem?
+	type IsCallable: FilterStack<<Self as Trait>::Call>;
+}
+
+/// A global extrinsic index, formed as the extrinsic index within a block, together with that
+/// block's height. This allows a transaction in which a multisig operation of a particular
+/// composite was created to be uniquely identified.
+#[derive(Copy, Clone, Eq, PartialEq, Encode, Decode, Default, RuntimeDebug)]
+pub struct Timepoint<BlockNumber> {
+	/// The height of the chain at the point in time.
+	height: BlockNumber,
+	/// The index of the extrinsic at the point in time.
+	index: u32,
+}
+
+/// An open multisig operation.
+#[derive(Clone, Eq, PartialEq, Encode, Decode, Default, RuntimeDebug)]
+pub struct Multisig<BlockNumber, Balance, AccountId> {
+	/// The extrinsic when the multisig operation was opened.
+	when: Timepoint<BlockNumber>,
+	/// The amount held in reserve of the `depositor`, to be returned once the operation ends.
+	deposit: Balance,
+	/// The account who opened it (i.e. the first to approve it).
+	depositor: AccountId,
+	/// The approvals achieved so far, including the depositor. Always sorted.
+	approvals: Vec<AccountId>,
+}
+
+decl_storage! {
+	trait Store for Module<T: Trait> as Utility {
```

Comment:
```suggestion
	trait Store for Module<T: Trait> as Multisig {
```

We would need a migration if we wanted to do this...

Reply:
I assume the rest of this file is copy paste, so not really reviewing it